src/ClaimValue.php

```diff
@@ -0,0 +1,44 @@
+<?php
+namespace Lcobucci\JWT;
+
+use InvalidArgumentException;
+use JsonSerializable;
+
+final class ClaimValue implements JsonSerializable
+{
+    private $value;
+
+    public function __construct($value)
+    {
+        $this->ensureValueIsJsonSerializable($value);
+
+        $this->value = $value;
```

Comment:
Instead of ensuring, just serialize/deserialize it (de-references the value: good)